plugins/infiniteSky.cpp

```diff
@@ -95,25 +95,68 @@ void doInfiniteSky(color_ostream& out, int32_t howMany) {
     CoreSuspender suspend;
     int32_t x_count_block = world->map.x_count_block;
     int32_t y_count_block = world->map.y_count_block;
-    for ( int32_t count = 0; count < howMany; count++ ) {
-        //change the size of the pointer stuff
-        int32_t z_count_block = world->map.z_count_block;
-        df::map_block**** block_index = world->map.block_index;
-        for ( int32_t a = 0; a < x_count_block; a++ ) {
-            for ( int32_t b = 0; b < y_count_block; b++ ) {
-                df::map_block** blockColumn = new df::map_block*[z_count_block+1];
-                memcpy(blockColumn, block_index[a][b], z_count_block*sizeof(df::map_block*));
-                blockColumn[z_count_block] = NULL;
-                delete[] block_index[a][b];
-                block_index[a][b] = blockColumn;
-
-                //deal with map_block_column stuff even though it'd probably be fine
-                df::map_block_column* column = world->map.column_index[a][b];
-                if ( !column ) {
-                    out.print("%s, line %d: column is null (%d, %d).\n", __FILE__, __LINE__, a, b);
+    int32_t z_count_block = world->map.z_count_block;
+    df::map_block ****block_index = world->map.block_index;
+
+    for (int32_t a = 0; a < x_count_block; a++) {
+        for (int32_t b = 0; b < y_count_block; b++) {
+            // Allocate a new block column and copy over data from the old
+            df::map_block **blockColumn =
+                new df::map_block *[z_count_block + howMany];
+            memcpy(blockColumn, block_index[a][b],
+                   z_count_block * sizeof(df::map_block *));
+            delete[] block_index[a][b];
+            block_index[a][b] = blockColumn;
+
+            df::map_block *last_air_block = blockColumn[z_count_block - 1];
+            for (int32_t count = 0; count < howMany; count++) {
+                df::map_block *air_block = new df::map_block();
+                std::fill(&air_block->tiletype[0][0],
+                          &air_block->tiletype[0][0] + (16 * 16),
+                          df::tiletype::OpenSpace);
+
+                // Set block positions properly (based on prior air layer)
+                air_block->map_pos = blockColumn[z_count_block - 1]->map_pos;
```

Comment:
Any reason for not using `last_air_block` here (and 2 lines down)?

Reply:
No reason, I simply forgot to update those lines.